Samples/iOS-Swift/iOS-SwiftUITests/LaunchUITests.swift

```diff
@@ -80,7 +81,48 @@ class LaunchUITests: BaseUITest {
 
     func testCheckSlowAndFrozenFrames() {
         app.buttons["Extra"].tap()
-        checkSlowAndFrozenFrames()
+        
+        let (totalFrames, slowFrames, frozenFrames) = getFramesStats()
+       
+        let slowFramesPercentage = Double(slowFrames) / Double(totalFrames)
+        expect(slowFramesPercentage).to(beLessThan(0.5), description: "Too many slow frames.")
+        
+        let frozenFramesPercentage = Double(frozenFrames) / Double(totalFrames)
+        expect(frozenFramesPercentage).to(beLessThan(0.5), description: "Too many frozen frames.")
+    }
+    
+    func testCheckTotalFrames() {
+        app.buttons["Extra"].tap()
+        
+        let (startTotalFrames, _, _) = getFramesStats()
+        let startDate = Date()
+    
+        let dispatchQueue = DispatchQueue(label: "CheckSlowAndFrozenFrames")
+        let dispatchGroup = DispatchGroup()
+        dispatchGroup.enter()
+        
+        dispatchQueue.asyncAfter(deadline: .now() + 3.0) {
+            dispatchGroup.leave()
+        }
+        dispatchGroup.wait()
+        
+        let (endTotalFrames, _, _) = getFramesStats()
+        let endDate = Date()
+        
+        let secondsBetween = endDate.timeIntervalSince(startDate)
+        
+        // Retrieving the number of frames from the text label is inaccurate.
+        // Therefore, we add a one-second tolerance for the expected amount of frames.
+        // We only want to validate if the number of total frames is entirely off.
+        let maximumFramesPerSecond = Double(UIScreen.main.maximumFramesPerSecond)
+        
+        let expectedMinimumTotalFrames = (secondsBetween - 1.0) * maximumFramesPerSecond
+        let expectedMaximumTotalFrames = (secondsBetween + 1.0) * maximumFramesPerSecond
+        
+        let actualTotalFrames = Double(endTotalFrames - startTotalFrames)
+        
+        expect(actualTotalFrames) >= expectedMinimumTotalFrames
```

Comment:
This is kinda weird codewise speaking. 
A function call that is compared to a value, but the comparison result is "not used", so weird.

Not advocating for changing back to not use nimble. 
Just that, this is the first time I actually seeing this being used, it was weird.

Reply:
with Nimble you can also write it like this

> expect(actual).to(beGreaterThanOrEqualTo(expected))

but I prefer the shorthand, even though it needs a bit getting used to